docs/tutorials/01_KalmanFilterTutorial.py

```diff
@@ -0,0 +1,415 @@
+#           \textbf{h}(\textbf{z}_{k}\ |\ \textbf{x}_{k},\ \textbf{v}_{k})
+#
+# which describes how the sensor (providing us with measurments) responds to the state
+# :math:`\textbf{x}_{k}` of the object (at time :math:`k`), with some noise :math:`\textbf{v}_{k}`
+# (since our measurements come with their own uncertainties).
+#
+#
+# We can rewrite:
+#
+# .. math::
+#           \textbf{w}_{k} &\sim \textbf{H}_{k}(\eta_{1}, \eta_{2}, ...)\\
+#           \textbf{v}_{k} &\sim \textbf{Z}_{k}(\zeta_{1}, \zeta_{2}, ...)
+#
+# where :math:`\textbf{H}` and :math:`\textbf{Z}` are stochastic processes (something
+# demonstrating randomness in output) with parameters :math:`\eta` and :math:`\zeta`.
+# These functions generate instances of the noises represented by :math:`\textbf{w}` and
+# :math:`\textbf{v}`.
+#
+# So,
+#
+# .. math::
+#           \textbf{x}_{k|k-1} &\sim p(\textbf{x}_{k}\ |\ \textbf{x}_{k-1},\ \eta_{k})\\
+#               \textbf{z}_{k} &\sim p(\textbf{z}_{k}\ |\ \textbf{x}_{k},\ \zeta_{k})
+#
+# In other words, we concern ourselves with the probability distribution of the state and
+# measurement.
+#
+# Then, adapting the law of total probability and Baye's rule:
+#
+# .. math::
+#           p(x) &= \int_{X} p(x|x')p(x')dx'\\
+#           P(X|Y) &= \frac{P(Y|X)P(X)}{P(Y)}
+#
+# We get:
+#
+# The Chapman-Kolmogorov equation, which defines the probability
+# :math:`p(\textbf{x}_{k}\ |\ \textbf{z}_{1:k-1})` of the current state given that all observations
+# up to the previous step :math:`k-1` have occurred):
+#
+# .. math::
+#           p(\textbf{x}_{k}\ |\ \textbf{z}_{1:k-1}) =
+#           \int_{-\infty}^{\infty} p(\textbf{x}_{k}\ |\ \textbf{x}_{k-1})
+#           p(\textbf{x}_{k-1}\ |\ \textbf{z}_{1:k-1})\ d\textbf{x}_{k-1}
+#
+# and the probability :math:`p(\textbf{x}_{k}\ |\ \textbf{z}_{1:k})` of the current state given all
+# observations up to (time) :math:`k`:
+#
+# .. math::
+#           p(\textbf{x}_{k}\ |\ \textbf{z}_{1:k}) = \frac{p(\textbf{z}_{k}\ |\ \textbf{x}_{k})
+#                                                          p(\textbf{x}_{k}\ |\ \textbf{z}_{1:k-1})}
+#                                                         {p(\textbf{z}_{k})}
+#
+#
+# Our state estimation process runs recursively: Given a *prior* estimate of a state vector, we
+# *predict* the state at the next time step, and *update* this with a measurement to get a
+# *posterior* state estimate.
+
+#%%
+# **The following is a derivation of the anayltic solution to the recursive equations above,
+# alongside a demonstration of the Kalman Filter within Stone Soup:**
+
+# Some general imports and set up
+from datetime import datetime
+from datetime import timedelta
+
+import numpy as np
+
+# Figure to plot truth (and future data)
+from matplotlib import pyplot as plt
+
+#%%
+# Simulate target
+# -----------------
+#
+# Consider the case of a target having state:
+#
+# .. math::
+#       \vec{x} = \begin{bmatrix} x\\ v_{x}\\ y\\ v_{y} \end{bmatrix}
```

Comment:
just a notation comment here, I prefer \dot(x) rather than v_{x} to represent velocity. I guess we just need to set out a consistent vision for this for all notebooks and tutorials.